src/porepy/models/applications/terzaghi_model.py

```diff
@@ -0,0 +1,706 @@
+            self.numerical_nondim_pressure = np.ones(sd.num_cells)
+            self.exact_nondim_pressure = np.ones(sd.num_cells)
+        else:
+            self.numerical_pressure = data[pp.STATE][p_var]
+            self.exact_pressure = model.exact_pressure(t)
+            self.numerical_nondim_pressure = model.nondim_pressure(
+                self.numerical_pressure
+            )
+            self.exact_nondim_pressure = model.nondim_pressure(self.exact_pressure)
+
+        # Mechanical variables
+        if t == 0:
+            self.numerical_displacement = np.zeros(sd.dim * sd.num_cells)
+            self.numerical_consolidation_degree = 0.0
+            self.exact_consolidation_degree = 0.0
+        else:
+            self.numerical_displacement = data[pp.STATE][u_var]
+            self.numerical_consolidation_degree = model.numerical_consolidation_degree(
+                self.numerical_displacement, self.numerical_pressure
+            )
+            self.exact_consolidation_degree = model.exact_consolidation_degree(t)
+
+        # Error variables
+        self.pressure_error = model.l2_relative_error(
+            sd=sd,
+            true_val=self.exact_pressure,
+            approx_val=self.numerical_pressure,
+            is_scalar=True,
+            is_cc=True,
+        )
+        self.consolidation_degree_error = np.abs(
+            self.exact_consolidation_degree - self.numerical_consolidation_degree
+        )
+
+
+class Terzaghi(pp.ContactMechanicsBiot):
+    """Parent class for Terzaghi's consolidation problem model.
+
+    Examples:
+
+        .. code:: Python
+
+        # Import modules
+        import porepy as pp
+        from time import time
+
+        # Time manager
+        time_manager = pp.TimeManager([0, 0.01, 0.1, 0.5, 1, 2], 0.001, constant_dt=True)
+
+        # Model parameters
+        params = {
+                'alpha_biot': 1.0,  # [-]
+                'height': 1.0,  # [m]
+                'lambda_lame': 1.65E9,  # [Pa]
+                'mu_lame': 1.475E9,  # [Pa]
+                'num_cells': 20,
+                'permeability': 9.86E-14,  # [m^2]
+                'perturbation_factor': 1E-6,
+                'plot_results': True,
+                'specific_weight': 9.943E3,  # [Pa * m^-1]
+                'time_manager': time_manager,
+                'upper_limit_summation': 1000,
+                'use_ad': True,
+                'vertical_load': 6E8,  # [N * m^-1]
+                'viscosity': 1E-3,  # [Pa * s]
+            }
+
+        # Run model
+        tic = time()
+        print("Simulation started...")
+        model = Terzaghi(params)
+        pp.run_time_dependent_model(model, params)
+        print(f"Simulation finished in {round(time() - tic)} sec.")
+
+    """
+
+    def __init__(self, params: dict):
+        """Constructor of the Terzaghi class.
+
+        Args:
+            params: Model parameters. Required entries are:
+
+                - 'alpha_biot' : Biot constant (int or float).
```

Comment:
The general approach has been to not require this many parameters, but rather provide unitary defaults. However, I think this is acceptable for now, since it will be thoroughly reworked when model_refactoring is merged

Reply:
After a conversation with Eirik, we agreed on providing default parameters rather than making them mandatory. However, unlike standard models, default values for applications should be physically sounded and not unitary. Are you OK with this @IvarStefansson?